### PR TITLE
Removed redundant _texts variables.

### PR DESCRIPTION
**Pull request by awhitford**
* **Colorize**
  * Added a `_timer` variable so that it can be properly _canceled_ during `dispose`.
  * Updated code to use proper Character counting so that it is Emoji friendly.
* **Typewriter**
  * Corrected code to use proper Character counting so that it is Emoji friendly.
* **General**
  * Added types to Animation declarations.
  * Removed the redundant `_texts` variables.
  * Updated code to satisfy _pedantic_ rules.

### CODE REVIEW

File: lib/src/typewriter.dart [hunk 1]
```diff
@@ -236,21 +231,23 @@ class _TypewriterState extends State<TypewriterAnimatedTextKit>
...
       vsync: this,
     );
 
-    _typewriterText =
-        StepTween(begin: 0, end: _texts[_index]['text'].length + 8)
-            .animate(_controller)
-              ..addStatusListener(_animationEndCallback);
+    _typewriterText = StepTween(
+      begin: 0,
+      end: textLen + 8,
```
Comment (awhitford) [inline lib/src/typewriter.dart:242]: Does anybody know why 8 is being added?  What is significant about 8?

**Comment by awhitford** [comment]
Investigating #128 led me to realize that `TextLiquidFill` has a bug where the wave animation continues to repeat even after the load has completed, so I added a fix to stop that.

File: lib/src/typewriter.dart (hunk 1 above)
Comment (aagarwal1012) [reply lib/src/typewriter.dart:242]: `8` is added for the `_` typing animation after the given text is animated or typed.

**Comment by aagarwal1012** [comment]
@awhitford, your suggested changes look good to me, great work! 💯

**Review by aagarwal1012 — APPROVED** [review]

File: lib/src/typewriter.dart (hunk 1 above)
Comment (awhitford) [reply lib/src/typewriter.dart:242]: Why 8 and not 1?

Comment (aagarwal1012) [reply lib/src/typewriter.dart:242]: Idk it's a random number that I added at the time of development, a lot of code has been changed after that.

**Pull request merged**